src/util.c

```diff
@@ -0,0 +1,41 @@
+#define _GNU_SOURCE
+#include <stdarg.h>
+#include <stdbool.h>
+#include <stdio.h>
+#include <string.h>
+
+static __thread bool nm_err_state                       = false;
+static __thread char nm_err_buf[2048]                   = {0};
+static __thread char nm_err_buf_tmp[sizeof(nm_err_buf)] = {0}; // in case the format string overlaps
```

Comment:
Does this one need to be static & thread-local? Couldn't it simply be function-scope in `nm_err_set`?

Reply:
I wanted to keep it close to prevent mishaps with the size, and I also didn't want to use up stack space unnecessarily.